backend/dataall/modules/maintenance/api/resolvers.py

```diff
@@ -0,0 +1,26 @@
+from dataall.base.api.context import Context
+from dataall.core.permissions.services.tenant_policy_service import TenantPolicyValidationService
+from dataall.modules.maintenance.api.enums import MaintenanceModes
+from dataall.modules.maintenance.api.types import Maintenance
+from dataall.modules.maintenance.services.maintenance_service import MaintenanceService
+
+
+def start_maintenance_window(context: Context, source: Maintenance, mode: str):
+    """Starts the maintenance window"""
+    if mode not in [item.value for item in list(MaintenanceModes)]:
+        raise Exception('Mode is not conforming to the MaintenanceModes enum')
+    # Check from the context if the groups contains the DAAAdminstrators group
+    if not TenantPolicyValidationService.is_tenant_admin(context.groups):
```

Comment:
Just curious, any particular reason this should be done in services .Is it because of the use of `TenantPolicyValidationService ` which inherently is a service and thus service calls should be in the service layer ? Asking this because we validate before calling the Service layers , like for e.g. in create_dataset method, we call the `RequestValidator.validate_creation_request(input)` before calling the services layer functions

Reply:
@dlpzx , I have moved this to the services

Reply:
It is just an application agreement. All business logic, which includes data.all custom permissions are part of the service layer